lib/elixir/lib/macro/env.ex

```diff
@@ -185,6 +185,39 @@ defmodule Macro.Env do
   def fetch_alias(%{__struct__: Macro.Env, aliases: aliases}, atom) when is_atom(atom),
     do: Keyword.fetch(aliases, :"Elixir.#{atom}")
 
+  @doc """
+  Fetches the name that the given module or atom has been aliased as.
+
+  Returns `{:ok, name}` if the alias exists, `:error` otherwise.
+
+  ## Examples
+
+      iex> alias Foo.Bar, as: Baz
+      iex> Baz
+      Foo.Bar
+      iex> Macro.Env.fetch_aliased_as(__ENV__, Foo.Bar)
+      {:ok, :Baz}
+      iex> Macro.Env.fetch_aliased_as(__ENV__, :"Foo.Bar")
+      {:ok, :Baz}
+      iex> Macro.Env.fetch_aliased_as(__ENV__, Foo.Unknown)
+      :error
+
+  """
+  @doc since: "1.15.0"
+  @spec fetch_aliased_as(t, atom) :: {:ok, atom} | :error
+  def fetch_aliased_as(%{__struct__: Macro.Env, aliases: aliases}, atom) when is_atom(atom) do
+    qualified =
+      case Macro.classify_atom(atom) do
+        :alias -> atom
+        _ -> :"Elixir.#{atom}"
+      end
```

Comment:
We should not do this because we can do `alias :foo, as: Omg`. We should expect an atom and not do any transformation. :)